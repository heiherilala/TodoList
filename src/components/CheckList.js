export default function CheckList(props) {
  return (
    <div className="form-check form-switch row">
      {props.checked&&<input
        type="checkbox"
        checked={false}
        className="col-sm-1"
        id={props.id}
        name={props.name}
        value={props.value}
        onChange={
          e=>{
            if (e.target.checked) {
              props.functionClick()
              e.target.checked=false
            }
          }
        }
      />}
      <label className="col-sm-10 text-white" htmlFor={props.id} value={props.value} > {props.value} </label>
      <br />
    </div>
  );
}
