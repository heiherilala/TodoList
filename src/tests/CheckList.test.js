import CheckList from "../components/CheckList";
import { render, fireEvent, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
const bodyTextMock = "Body text moxk."

    test('Item has a correct value', () => {
        render (<CheckList
            value={bodyTextMock}
            id={"todoID"}
            functionClick={()=>{}}
            checked={true}
        />)
        const item = screen.getByRole("checkbox");
        expect(item.value).toBe(bodyTextMock);
    });

    test('Item has checkbox and its checked = false on creation if checked in checklist is true.', () => {
        render (<CheckList
            value={bodyTextMock}
            id={"todoID"}
            functionClick={()=>{}}
            checked={true}
            name="todo"
        />)
        expect(screen.queryByRole("checkbox")).not.toBeNull();
        const item = screen.getByRole("checkbox");
        expect(item.checked).toBeFalsy();
    });

    test("Item does not have checkbox on creation when checked in Checklist is false.", () => {
        render(
            <CheckList
            value={bodyTextMock}
            id={"todoID"}
            functionClick={()=>{}}
            checked={false}
            name="done"
            />
        );
        expect(screen.queryByRole("checkbox")).toBeNull();
    });

    test("function work when clecked", () => {
        const functionMock = jest.fn();
        render(
            <CheckList
            value={bodyTextMock}
            id={"todoID"}
            functionClick={functionMock}
            checked={true}
            />
        );
        const ckeckboxItem = screen.getByRole("checkbox");
        fireEvent.click(ckeckboxItem);
        expect(functionMock.mock.calls.length).toBe(1);
    });

    it("have to be same as the Snapshot", () => {
        const component = renderer.create(
                <CheckList
                    value={bodyTextMock}
                    id={"todoID"}
                    functionClick={()=>{}}
                    checked={true}
                />
            );
      
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });