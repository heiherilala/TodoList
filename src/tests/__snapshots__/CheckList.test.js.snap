// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`have to be same as the Snapshot 1`] = `
<div
  className="form-check form-switch row"
>
  <input
    checked={false}
    className="col-sm-1"
    id="todoID"
    onChange={[Function]}
    type="checkbox"
    value="Body text moxk."
  />
  <label
    className="col-sm-10 text-white"
    htmlFor="todoID"
    value="Body text moxk."
  >
     
    Body text moxk.
     
  </label>
  <br />
</div>
`;
